Remotion/Data/DomainObjects/ReflectionUtility.cs

```diff
@@ -262,10 +267,24 @@ public static Type GetObjectListTypeParameter (Type type)
     {
       ArgumentUtility.CheckNotNull ("type", type);
 
-      var typeParameters = type.GetAscribedGenericArguments (typeof (ObjectList<>));
-      var typeParameter = typeParameters[0];
+      var (_, typeParameter) = s_canAscribeToGenericArgumentsOfObjectListCache.AddOrUpdate (
+          type,
+          static t => ValueTuple.Create (
+              t.CanAscribeTo (typeof (ObjectList<>)),
+              t.GetAscribedGenericArguments (typeof (ObjectList<>))[0]),
+          static (k, v) =>
+          {
+            if (v.Item2 is not null)
+              return v;
+
+            return ValueTuple.Create (
+                v.Item1,
+                k.GetAscribedGenericArguments (typeof (ObjectList<>))[0]);
```

Comment:
I can't populate the cache in one run, because some tests require `GetAscribedGenericArguments` to throw an exception in this method (`GetObjectListTypeParameter`), which only happens if `canAscribe` is `false`, therefore I can't guard this with an `if (canAscribe)` like i did in `IsObjectList` on line `160` because then it will return `null` for `Item2` aka `ItemType` and the test would fail.

Reply:
The exception can be thrown after we test for canascribe==false from the cache. We only need to ensure that nothing bad happens, not that the exact same exceptions happen.